src/coreclr/jit/compiler.cpp

```diff
@@ -5138,26 +5138,20 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
         m_llvm->Lower();
     });
 
-    if (opts.OptimizationEnabled())
-    {
-        // When optimizing, we'll sort the locals on the shadow stack by ref count.
-        lvaMarkLocalVars();
-    }
-
-    DoPhase(this, PHASE_ALLOCATE_SHADOW_STACK, [this]() {
-        m_llvm->Allocate();
-    });
-
     fgSsaDomTree = nullptr;
     if (opts.OptimizationEnabled())
     {
         DoPhase(this, PHASE_BUILD_SSA, [this]() {
-            lvaMarkLocalVars();
+            lvaComputeRefCounts(/* isRecompute */ true, /* setSlotNumbers */ false);
```

Comment:
`lvaMarkLocalVars` is really meant to only be called once, `lvaComputeRefCounts` is what regular lowering calls.